### PR TITLE
Changed All Users button to Users per the user story specs, added  us…

### PR DESCRIPTION
**Pull request by GDemps**
…ers_controller, users/index.html.erb, and test in /admin/users/index_spec.rb

[Reference story or milestone being closed]

**Contributors**
- Driver(s): Grant
  
**What was changed:**
Fixed route issues and file location issues with first PR
Added admin/users_controller with index method, test: admin/users/index_spec.rb, and view admin/users/index.html.erb

disregard other changes resulting from my nuke of local and github branches and re-pulling

**Any tech notes (if applicable)**
Please look at where my line comments are.  The rest is from my git fire and should be fine considering I pulled them down from development. Cheers!

### CODE REVIEW

File: app/controllers/admin/base_controller.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class Admin::BaseController < ApplicationController
```
Comment (GDemps) [inline app/controllers/admin/base_controller.rb:1]: controller that I added.

File: app/controllers/admin/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class Admin::UsersController < Admin::BaseController
```
Comment (GDemps) [inline app/controllers/admin/users_controller.rb:1]: Added this controller

Comment (LHJE) [reply app/controllers/admin/users_controller.rb:1]: Looks awesome!

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -10,10 +10,30 @@
...
-      <%= link_to "Cart: #{cart.total_items}", "/cart" %>
+
+      <% if !current_user %>
+        <%= link_to "Register", "/register"%>
+        <%= link_to "Login", "/login" %>
+        <%= link_to "Cart: #{cart.total_items}", "/cart" %>
+      <% else %>
+        <% if current_user.role == "admin" %>
+          <%= link_to "Admin Dashboard", "/admin" %>
+          <%= link_to "Users", "/admin/users" %>
```
Comment (GDemps) [inline app/views/layouts/application.html.erb:25]: Changed to "Users" from "All Users" per the user story direction.

File: config/routes.rb [hunk 4]
```diff
@@ -1,12 +1,17 @@
...
+    get "/merchant", to: "dashboard#index", as: :dashboard
+  end
 
   get "/merchants", to: "merchants#index"
   get "/merchants/new", to: "merchants#new"
   get "/merchants/:id", to: "merchants#show"
   post "/merchants", to: "merchants#create"
   get "/merchants/:id/edit", to: "merchants#edit"
-  patch "/merchants/:id", to: "merchants#update"
+  patch "/merchants/:id", to: "merchants#update", as: :merchant_update
```
Comment (LHJE) [inline config/routes.rb:14]: Nice!  Much more specific.

File: config/routes.rb [hunk 5]
```diff
@@ -35,6 +42,28 @@
...
+  get "profile/edit", to: "users#edit"
+  patch "profile/edit", to: "users#update"
+  get "/profile/orders", to: "users#order"
+
+  get "/login", to: "sessions#new"
+  post "/login", to: "sessions#create"
+
+  get "/logout", to: "sessions#destroy"
+
+  get "/admin/users", to: "admin/users#index"
```
Comment (GDemps) [inline config/routes.rb:57]: this is the route added

File: config/routes.rb [hunk 6]
```diff
@@ -35,6 +42,28 @@
...
+
+
+  namespace :admin do
+    get "/admin", to: "dashboard#index", as: :dashboard
+  end
+
+  scope :admin do
```
Comment (LHJE) [inline config/routes.rb:64]: Nice use of scope!

File: spec/features/admin/users/index_spec.rb [hunk 7]
```diff
@@ -0,0 +1,33 @@
+#spec/features/admin/users/index_spec.rb
```
Comment (GDemps) [inline spec/features/admin/users/index_spec.rb:1]: Added this test

File: spec/features/admin/users/index_spec.rb [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+      expect(page).to have_link("Leah")
+      expect(page).to have_link("Grant")
+      expect(page).to have_link("Luke Hunter James-Erickson")
+      expect(current_path).to eq("/admin/users")
+      expect(page).to have_content("#{@admin_user.created_at}")
+      expect(page).to have_content("#{@reg_user.created_at}")
+      expect(page).to have_content("#{@merchant_employee.created_at}")
+      expect(page).to have_content("User Role: #{@admin_user.role}")
+
+    end
```
Comment (LHJE) [inline spec/features/admin/users/index_spec.rb:31]: All of this looks stupendous!

**Review by LHJE — APPROVED** [review]
Great job Grant, this one was a bear to be sure.

File: app/views/admin/users/index.html.erb [hunk 9]
```diff
@@ -0,0 +1,6 @@
+<h1>Users</h1>
```
Comment (GDemps) [inline app/views/admin/users/index.html.erb:1]: This is the view I added for admin/users/index.html.erb

**Pull request merged**